.github/workflows/build-and-push.yml

```diff
@@ -1,99 +1,42 @@
-        with:
-          registry: quay.io
-          username: ${{ secrets.QUAY_IMAGE_SCLORG_BUILDER_USERNAME }}
-          password: ${{ secrets.QUAY_IMAGE_SCLORG_BUILDER_TOKEN }}
-
-      - name: Check if .exclude-${{ matrix.tag }} is present in version directory
-        id: check_exclude_file
-        # https://github.com/marketplace/actions/file-existence
-        uses: andstor/file-existence-action@v1
-        with:
-          files: "${{ matrix.version }}/.exclude-${{ matrix.tag }}"
-
-      - name: Build image
-        if: steps.check_exclude_file.outputs.files_exists == 'false'
-        id: build-image
-        # https://github.com/marketplace/actions/buildah-build
-        uses: redhat-actions/buildah-build@v2
-        with:
-          dockerfiles: ${{ matrix.version }}/${{ matrix.dockerfile }}
-          image: ${{ steps.base-image-name.outputs.image_name}}-${{ steps.short_version.outputs.SHORT_VER }}-${{ matrix.tag }}
-          context: ${{ matrix.version }}
-          tags: latest ${{ matrix.tag }} ${{ github.sha }}
-
-      - name: Push image to Quay.io/${{ matrix.registry_namespace }} namespace
-        if: ${{ steps.check_exclude_file.outputs.files_exists == 'false' && matrix.registry_namespace == 'centos7' }}
-        id: push-to-quay-centos7
-        uses: redhat-actions/push-to-registry@v2.2
+      - name: Build and push to quay.io registry
+        uses: sclorg/build-and-push-action@v1
         with:
-          image: ${{ steps.build-image.outputs.image }}
-          tags: ${{ steps.build-image.outputs.tags }}
-          registry: quay.io/${{ matrix.registry_namespace }}
-          username: ${{ secrets.QUAY_IMAGE_BUILDER_USERNAME }}
-          password: ${{ secrets.QUAY_IMAGE_BUILDER_TOKEN }}
-
-      - name: Push image to Quay.io/${{ matrix.registry_namespace }} namespace
-        if: ${{ steps.check_exclude_file.outputs.files_exists == 'false' && matrix.registry_namespace != 'centos7' }}
-        id: push-to-quay-sclorg
-        uses: redhat-actions/push-to-registry@v2.2
-        with:
-          image: ${{ steps.build-image.outputs.image }}
-          tags: ${{ steps.build-image.outputs.tags }}
-          registry: quay.io/${{ matrix.registry_namespace }}
-          username: ${{ secrets.QUAY_IMAGE_SCLORG_BUILDER_USERNAME }}
-          password: ${{ secrets.QUAY_IMAGE_SCLORG_BUILDER_TOKEN  }}
-
-      - name: Print image url
-        if: ${{ matrix.registry_namespace == 'centos7' }}
-        run: echo "Image pushed to ${{ steps.push-to-quay-centos7.outputs.registry-paths }}"
-
-      - name: Print image url
-        if: ${{ matrix.registry_namespace != 'centos7' }}
-        run: echo "Image pushed to ${{ steps.push-to-quay-sclorg.outputs.registry-paths }}"
+          registry: "quay.io"
+          registry_namespace: ${{ matrix.registry_namespace }}
+          registry_username: ${{ secrets[matrix.quayio_username] }}
+          registry_token: ${{ secrets[matrix.quayio_token] }}
```

Comment:
I thinks the format should be:
${{ secrets.[matrix.quayio_token] }} right?
I guess, '.' is missing.

Reply:
I was inspired by [this article](https://sbulav.github.io/terraform/github-actions-matrix-secrets/).
The dot is not used there.

Reply:
I see. ok.